utils/frame/sub-du/src/main.rs

```diff
@@ -0,0 +1,271 @@
+			write!(f, "{} {} {}\n", Size(s.size), get_prefix(3), s)?;
+		}
+		Ok(())
+	}
+}
+
+impl Pallet {
+	fn new(name: String) -> Self {
+		Self { name, ..Default::default() }
+	}
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum StorageItem {
+	Value(usize),
+	Map(usize, usize),
+}
+
+impl std::fmt::Display for StorageItem {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		match self {
+			Self::Value(bytes) => write!(f, "Value({} bytes)", bytes.separated_string()),
+			Self::Map(bytes, count) => {
+				write!(f, "Map({} bytes, {} keys)", bytes.separated_string(), count)
+			},
+		}
+	}
+}
+
+impl Default for StorageItem {
+	fn default() -> Self {
+		Self::Value(0)
+	}
+}
+
+#[derive(Debug, Clone, Default)]
+struct Storage {
+	pub name: String,
+	pub size: usize,
+	pub item: StorageItem,
+}
+
+impl std::fmt::Display for Storage {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let item_style = Style::new().italic();
+		write!(f, "{} => {}", item_style.paint(self.name.clone()), self.item)
+	}
+}
+
+impl Storage {
+	fn new(name: String, item: StorageItem) -> Self {
+		let size = match item {
+			StorageItem::Value(s) => s,
+			StorageItem::Map(s, _) => s,
+		};
+		Self { name, item, size }
+	}
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+	name = "sub-du",
+	about = "a du-like tool that prints the map of storage usage of a substrate chain"
+)]
+struct Opt {
+	/// The block number at which the scrap should happen. Use only the hex value, no need for a
+	/// `0x` prefix.
+	#[structopt(long)]
+	at: Option<Hash>,
+
+	/// The node to connect to.
+	#[structopt(long, default_value = "wss://rpc.polkadot.io:443")]
+	uri: String,
+
+	/// If true, intermediate values will be printed.
+	#[structopt(long, short)]
+	progress: bool,
+
+	/// Weather to scrape all pairs or just the size of them.
+	///
+	/// If enabled, the command might take longer but then the number of keys in each map is also
+	/// scraped.
+	///
+	/// # Warning
+	///
+	/// This uses an unsafe RPC call and can only be used if the target node allows it.
+	#[structopt(long, short)]
+	scrape_pairs: bool,
+}
+
+/// create key prefix for a module as vec bytes. Basically twox128 hash of the given values.
+pub fn pallet_prefix_raw(module: &[u8], storage: &[u8]) -> Vec<u8> {
+	let module_key = twox_128(module);
+	let storage_key = twox_128(storage);
+	let mut final_key = Vec::with_capacity(module_key.len() + storage_key.len());
+	final_key.extend_from_slice(&module_key);
+	final_key.extend_from_slice(&storage_key);
+	final_key
+}
+
+type Block = RawBlock<ExtrinsicWrapper<Hash>>;
+
+#[tokio::main]
+async fn main() -> () {
+	tracing_subscriber::fmt::try_init().unwrap();
+	let opt = Opt::from_args();
+
+	let mut cfg = OnlineConfig::default();
+	cfg.transport = Transport::Uri(opt.uri);
+	cfg.transport.map_uri().await.unwrap();
+
+	// connect to a node.
+	let ext = remote_externalities::Builder::<Block>::new().mode(Mode::Online(cfg));
```

Comment:
yeah, I agree with Kian that we shouldn't use `remote_externalities` here instead use 

```
let rpc_client = substrate_rpc_client::ws_client(opt.uri).await.unwrap()
``` 

Then we don't need to have this `sp_testing::Block` which isn't really nice

Reply:
Yeah, the `sp_testing::Block` did not feel like an amazing approach. It's also kinda cryptic.